### PR TITLE
added enter children part of the application with adding multiple children, refactored test to match master to reduce conflicts

### PR DESCRIPTION
**Pull request by kurt-uk**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/secure/RapidBoard.jspa?rapidView=789&projectKey=FPL&view=detail&selectedIssue=FPL-7

### Change description ###
refactored tests to match master to reduce conflicts, added the page object for entering children and created the tests that would half fill in the form and complete the form with adding another child and saving. also added 2 chunk parallel testing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/pages/enterChildren/enterChildren.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			option: 'Yes'
+		},
+		child2SocialWorkerName: '#children_additionalChildren_0_enterChildren_socialWorkerName',
+		child2SocialWorkerTel: '#children_additionalChildren_0_enterChildren_socialWorkerTel',
+		child2AdditionalNeedsNo: '#children_additionalChildren_0_enterChildren_additionalNeeds-No',
+		child2ContactHiddenNo: '#children_additionalChildren_0_enterChildren_detailsHidden-No',
+		child2LitigationNo: '#children_additionalChildren_0_enterChildren_litigationIssues-No',
+	},
+
+	halfFillForm() {
```
Comment (geek-soft) [inline e2e/pages/enterChildren/enterChildren.js]: Page objects should expose API to the application page. That API should in readable manner communicate what actions are available on the page. Having methods like `halfFillForm` and `fillForm` does not communicate intent too well. It also makes it more difficult to reuse.

File: e2e/pages/enterChildren/enterChildren.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+
+const I = actor();
+
+module.exports = {
+
+	fields: {
```
Comment (geek-soft) [inline e2e/pages/enterChildren/enterChildren.js]: There is lots of duplication in field selectors between child 1 and child 2.

We could have just selector for child controls with some placeholder in the selector.

```
fields: (childNo) => {
  fullName: `#children_${childNo}_enterChildren_childName`,
  ...
}
```

Then we would use it in the following way:

```
const childSelectors = this.fields('firstChild');
I.fillField(childSelectors.fullName, 'Some name')
```

File: e2e/paths/enterChildren_test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const config = require('../config.js');
+
+Feature('enter children in application');
```
Comment (geek-soft) [inline e2e/paths/enterChildren_test.js]: Please start with capital letter ;)

File: e2e/pages/enterChildren/enterChildren.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+		enteredInformation: 'test',
```
[the remark this answers is not in the document]
Comment (geek-soft) [reply e2e/pages/enterChildren/enterChildren.js]: I would expose the following actions through this page object:
- enterChildDetails // name, DOB, gender
- defineChildSituation
- enterKeyDatesAffectingHearing
- enterSummaryOfCarePlan
- defineAdoptionIntention
- enterParentsDetails // mother name, father name, responsibility
- enterSocialWorkerDetails // name, phone
- defineChildAdditionalNeeds
- defineContactDetailsVisibility
- defineAbilityToTakePartInProceedings

File: codecept.json [hunk 5]
```diff
@@ -33,5 +34,10 @@
   },
   "tests": "./e2e/paths/*_test.js",
   "timeout": 10000,
-  "name": "ccd-docker"
-}
\ No newline at end of file
+  "name": "ccd-docker",
+  "multiple": {
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply codecept.json]: ok ill remove

File: e2e/pages/enterChildren/enterChildren.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+				addressOfChild: `#children_${childNo}_enterChildren_situationAddress`,
+			},
+			keyDates: `#children_${childNo}_enterChildren_keyDates`,
+			careAndContactPlan: `#children_${childNo}_enterChildren_careAndContact`,
+			adoptionNo: `#children_${childNo}_enterChildren_adoption-No`,
+			mothersName: `#children_${childNo}_enterChildren_mothersName`,
+			fathersName: `#children_${childNo}_enterChildren_fathersName`,
+			fatherResponsibleDropdown: {
+				selector: `#children_${childNo}_enterChildren_fathersResponsibility`,
+				option: `Yes`
```
Comment (swalker125) [inline e2e/pages/enterChildren/enterChildren.js]: still options in the fields

File: codecept.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (swalker125) [reply codecept.json]: still needs removing

File: e2e/pages/enterChildren/enterChildren.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+	enterChildDetails(name, day, month, year, gender = 'Boy') {
+		I.fillField(this.fields(activeChild).fullName, name);
+		I.fillField(this.fields(activeChild).DOB.day, day);
+		I.fillField(this.fields(activeChild).DOB.day, day);
+		I.fillField(this.fields(activeChild).DOB.month, month);
+		I.fillField(this.fields(activeChild).DOB.year, year);
+		I.selectOption(this.fields(activeChild).genderDropdown, gender);
+	}
+	,
```
Comment (swalker125) [inline e2e/pages/enterChildren/enterChildren.js]: comma is spaced down on a new line

File: e2e/pages/enterChildren/enterChildren.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+	defineChildSituation(day, month, year, situation = 'Living with respondents', address = '35 London Lane') {
+		I.selectOption(this.fields(activeChild).situationDropdown.selector, situation);
+		I.fillField(this.fields(activeChild).situationDropdown.dateStartedStaying.day, day);
+		I.fillField(this.fields(activeChild).situationDropdown.dateStartedStaying.month, month);
+		I.fillField(this.fields(activeChild).situationDropdown.dateStartedStaying.year, year);
+		I.fillField(this.fields(activeChild).situationDropdown.addressOfChild, address);
+	}
+	,
```
Comment (swalker125) [inline e2e/pages/enterChildren/enterChildren.js]: comma spacing

File: e2e/paths/enterChildren_test.js (hunk 3 above)
Comment (swalker125) [reply e2e/paths/enterChildren_test.js]: still needs doing

File: e2e/pages/enterChildren/enterChildren.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
+const I = actor();
+let activeChild = 'firstChild';
+
+module.exports = {
+
+	enteredInformation: 'test',
```
Comment (swalker125) [inline e2e/pages/enterChildren/enterChildren.js]: I can't find any usages of this when I cmd + f

**Comment by kurt-uk** [comment]
screen shot 2018-10-09 at 16 13 05
screen shot 2018-10-09 at 16 13 16
screen shot 2018-10-09 at 16 13 20
screen shot 2018-10-09 at 16 13 25
screen shot 2018-10-09 at 16 13 31
screen shot 2018-10-09 at 16 13 33
screen shot 2018-10-09 at 16 13 37
screen shot 2018-10-09 at 16 13 40

File: e2e/pages/enterChildren/enterChildren.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+	fields: (childNo) => {
+		return {
+			fullName: `#children_${childNo}_enterChildren_childName`,
+			DOB: {
+				day: `#children_${childNo}_enterChildren_childDOB-day`,
+				month: `#children_${childNo}_enterChildren_childDOB-month`,
+				year: `#children_${childNo}_enterChildren_childDOB-year`
+			},
+			genderDropdown: `#children_${childNo}_enterChildren_childGender`,
```
Comment (geek-soft) [inline e2e/pages/enterChildren/enterChildren.js]: In the future we should stop using control types in the selectors names.

File: e2e/pages/enterChildren/enterChildren.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+			socialWorkerTel: `#children_${childNo}_enterChildren_socialWorkerTel`,
+			additionalNeedsNo: `#children_${childNo}_enterChildren_additionalNeeds-No`,
+			contactHiddenNo: `#children_${childNo}_enterChildren_detailsHidden-No`,
+			litigationNo: `#children_${childNo}_enterChildren_litigationIssues-No`,
+		}
+	},
+	addChildButton: 'Add new',
+
+	addChild() {
+		I.click(this.addChildButton);
```
Comment (geek-soft) [inline e2e/pages/enterChildren/enterChildren.js]: We could do:

```
if (activeChild == 'additionalChildren_0') {
  throw new Error('Adding more children is not supported')
}
```

to reflect the fact API is limited.

**Pull request merged**